### DISCUSSION TITLE
Consider renaming "replica" to "device".

### DISCUSSION THREAD

**Discussion by ryneeverett** [GothenburgBitFactory/taskwarrior#3186] [Ideas]
What do we gain by introducing this new term "replica" which new users will have to learn and keep in mind while reading the documentation and configuring their systems? Wouldn't calling it a "device" eliminate the need for explanation and satisfy all the same purposes? Maybe "replica" makes a little more sense in the context of the code base, but I think the value of having the code base and the documentation terminology aligned probably outweighs that of having the ideal terminology in the code base -- users first!

**Comment by djmitche** [comment]
Is the term "device" used in the current documentation?

My concern with "device" is that I would like to encourage people to have purpose-specific replicas, maybe multiple per device. For example, applications like Bugwarrior would have their own replica that sync's with Taskwarrior's replica, rather than both operating on the same replica.

But, any name is not going to be perfectly clear, so maybe "device" (that may not always be a device) is clearer than "replica" (that may not be replicated).

**Reply by ryneeverett** [reply]
> Is the term "device" used in the current documentation?

Yes:

> Sync is a critical feature of TaskChampion, allowing users to consult and update the same task list on multiple devices, without requiring constant connection.

But I did not mean to imply that the current documentation has problems with mixing up this terminology, only that switching to an already-understood term would make documentation easier to digest.

> My concern with "device" is that I would like to encourage people to have purpose-specific replicas, maybe multiple per device. For example, applications like Bugwarrior would have their own replica that sync's with Taskwarrior's replica, rather than both operating on the same replica.

To clarify, it would be equally true to say that people have "purpose-specific clients", right? I'd considered that possibility but I assumed the "right" way to do such a thing was with tags and filters.

I think it's worth noting that with #3185 plus a stance that tags and filters are the right way to segment tasks, there is potentially no need for the sync server to have multi-client support at all.

> For example, applications like Bugwarrior would have their own replica that sync's with Taskwarrior's replica, rather than both operating on the same replica.

:exploding_head: I think this sentence might be evidence that "replica" is a difficult term. :smiley: I'm genuinely not sure how to parse this sentence. Bugwarrior's local replica syncs with taskwarrior's local replica? Bugwarrior's local replica syncs with the taskchampion sync server?
